tests/artifact_rest/test_api_factory.py

```diff
@@ -0,0 +1,115 @@
+# Copyright 2021 - 2023 Universität Tübingen, DKFZ, EMBL, and Universität zu Köln
+# for the German Human Genome-Phenome Archive (GHGA)
+#
+# Licensed under the Apache License, Version 2.0 (the "License");
+# you may not use this file except in compliance with the License.
+# You may obtain a copy of the License at
+#
+#     http://www.apache.org/licenses/LICENSE-2.0
+#
+# Unless required by applicable law or agreed to in writing, software
+# distributed under the License is distributed on an "AS IS" BASIS,
+# WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+# See the License for the specific language governing permissions and
+# limitations under the License.
+#
+
+"""Test the factory module."""
+
+
+import httpx
+import pytest
+from fastapi import FastAPI
+from hexkit.protocols.dao import DaoFactoryProtocol
+from hexkit.providers.mongodb.testutils import mongodb_fixture  # noqa: F401
+from hexkit.providers.mongodb.testutils import MongoDbFixture
+
+from metldata.artifacts_rest.api_factory import rest_api_factory
+from metldata.artifacts_rest.artifact_info import ArtifactInfo
+from tests.artifact_rest.test_load_artifacts import load_example_artifact_resources
+from tests.fixtures.artifact_info import EXAMPLE_ARTIFACT_INFOS, MINIMAL_ARTIFACT_INFO
+
+
+@pytest.mark.asyncio
+async def get_example_app_client(
+    dao_factory: DaoFactoryProtocol,
+    artifact_infos: list[ArtifactInfo] = EXAMPLE_ARTIFACT_INFOS,
+) -> httpx.AsyncClient:
+    """Return a test client for a FastAPI generated using the artifact_rest_factory."""
+
+    router = await rest_api_factory(
+        artifact_infos=artifact_infos, dao_factory=dao_factory
+    )
+
+    app = FastAPI()
+    app.include_router(router)
+    return httpx.AsyncClient(app=app, base_url="http://localhost:8080")
+
+
+@pytest.mark.asyncio
+async def test_artifacts_info_endpoint(mongodb_fixture: MongoDbFixture):  # noqa: F811
+    """Test happy path of using the artifacts info endpoint."""
+
+    expected_infos = EXAMPLE_ARTIFACT_INFOS
+
+    async with await get_example_app_client(
+        dao_factory=mongodb_fixture.dao_factory
+    ) as client:
+        response = await client.options("/artifacts")
+
+    response_json = response.json()
+    assert isinstance(response_json, list)
+    observed_infos = [ArtifactInfo(**info) for info in response_json]
+    assert observed_infos == expected_infos
+
+
+@pytest.mark.asyncio
+@pytest.mark.parametrize(
+    "artifact_name, expected_info",
+    [(info.name, info) for info in EXAMPLE_ARTIFACT_INFOS],
+)
```

Comment:
Here you would only use the name of the artifact in the parameter.

Reply:
see above